src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -98,12 +114,22 @@ internal static bool IsEnabled()
                 InjectHeaders(currentActivity, request);
             }
 
-            Task<HttpResponseMessage>? responseTask = null;
+            HttpResponseMessage? response = null;
+            TaskStatus taskStatus = TaskStatus.Faulted;
             try
             {
-                responseTask = base.SendAsync(request, cancellationToken);
-
-                return await responseTask.ConfigureAwait(false);
+                if (async)
+                {
+                    Task<HttpResponseMessage>? responseTask = base.SendAsync(request, cancellationToken);
```

Comment:
Why the question mark?

Reply:
No reason, it was a mistake.
Code changed due to following comments, so it's irrelevant now.